mesonbuild/mtest.py

```diff
@@ -1616,6 +1616,9 @@ def __init__(self, options: argparse.Namespace):
             for s in t.suite:
                 ss.add(s)
         self.suites = list(ss)
+        console_busy = any(runner.console_mode == ConsoleUser.GDB
+                           for runner in self.get_runners())
```

Comment:
I think it's better to only create the runners once.

I think what you want is to move this code:

```python
        if self.options.setup:
            env = self.merge_setup_options(options, test)
        else:
            env = os.environ.copy()
```

to `TestHarness.__init__`.